cmd/notary/delegations.go

```diff
@@ -188,6 +193,11 @@ func (d *delegationCommander) delegationAdd(cmd *cobra.Command, args []string) e
 		}
 	}
 
+	// If the user passes --all-paths, give the "" path
+	if allPaths {
```

Comment:
I agree that this is a nice compromise, `""` is a special prefix in this case and I agree it's better than having to add more object keys and booleans :)  @endophage and @cyli what do you think?

The only downside I can see here is if someone mistakenly adds `""` when they didn't mean to, and loses track of the previous paths, but this is a small headache and easily recoverable.